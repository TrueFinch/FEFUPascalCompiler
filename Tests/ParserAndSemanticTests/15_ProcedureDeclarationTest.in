program a;

procedure add1(const a, b: integer; var res: integer);
forward;

procedure add1(const a, b: integer; var res: integer);
begin
    res := a + b;
end;

begin
end.