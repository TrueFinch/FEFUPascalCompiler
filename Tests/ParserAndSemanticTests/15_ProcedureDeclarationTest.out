                                                                  a                                                                    
                                                                  ┼                                                                    
                                                              MainBlock                                                                
                       ┌──────────────────────────────────────────┴──────────────┬──────────────────────────────────────────┐          
                    ProcDecl                                                 ProcDecl                               CompoundStatement  
                       ┼                                               ┌─────────┴───────────────────────┐                             
                 CallableHeader                                  CallableHeader                   SubroutineBlock                      
  ┌─────────────┬──────┴─────────────┐            ┌─────────────┬──────┴─────────────┐                   ┼                             
 add1   FormalParamSection   FormalParamSection  add1   FormalParamSection   FormalParamSection  CompoundStatement                     
         ┌────┬─┴┬─────┐      ┌────┬─┴────┐              ┌────┬─┴┬─────┐      ┌────┬─┴────┐          ┌───┴                             
       const  a  b  integer  var  res  integer         const  a  b  integer  var  res  integer      :=                                 
                                                                                                  ┌──┴─┐                               
                                                                                                 res   +                               
                                                                                                      ┌┴─┐                             
                                                                                                      a  b                             
